src/main/kotlin/no/nav/familie/oppdrag/service/KonsistensavstemmingService.kt

```diff
@@ -10,42 +10,73 @@ import no.nav.familie.oppdrag.repository.UtbetalingsoppdragForKonsistensavstemmi
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
 import java.time.LocalDateTime
 
-
 @Service
 class KonsistensavstemmingService(
-        private val avstemmingSender: AvstemmingSender,
-        private val oppdragLagerRepository: OppdragLagerRepository,
+    private val avstemmingSender: AvstemmingSender,
+    private val oppdragLagerRepository: OppdragLagerRepository,
+    private val mellomlagringKonsistensavstemmingService: MellomlagringKonsistensavstemmingService,
 ) {
 
+    @Transactional
     fun utførKonsistensavstemming(request: KonsistensavstemmingUtbetalingsoppdrag) {
         utførKonsistensavstemming(request.fagsystem, request.utbetalingsoppdrag, request.avstemmingstidspunkt)
     }
 
     private fun utførKonsistensavstemming(
-            fagsystem: String,
-            utbetalingsoppdrag: List<Utbetalingsoppdrag>,
-            avstemmingstidspunkt: LocalDateTime,
+        fagsystem: String,
+        utbetalingsoppdrag: List<Utbetalingsoppdrag>,
+        avstemmingstidspunkt: LocalDateTime,
+        sendStartmelding: Boolean = true,
+        sendAvsluttmelding: Boolean = true
     ) {
-        val konsistensavstemmingMapper = KonsistensavstemmingMapper(fagsystem, utbetalingsoppdrag, avstemmingstidspunkt)
+        val metaInfo = KonsistensavstemmingMetaInfo(
+            Fagsystem.valueOf(fagsystem), avstemmingstidspunkt, sendStartmelding,
+            sendAvsluttmelding, utbetalingsoppdrag
+        )
+
+        if (metaInfo.erFørsteBatchIEnSplittetBatch()) {
+            mellomlagringKonsistensavstemmingService.nullstillMellomlagring(metaInfo)
+        }
+
+        val konsistensavstemmingMapper = opprettKonsistensavstemmingMapper(metaInfo)
```

Comment:
Bør ikke dette kun skje ved første melding?

Reply:
Kanske litt dumt navn; opretter et object for å samle alle variabler så det spares ikke i db.